tests/FiberTraces.py

```diff
@@ -125,528 +125,519 @@ def testFiberTraceFunctionFindingControl(self):
+                    self.assertAlmostEqual(oldestMask[i,j], mask[i,j])
+                    self.assertAlmostEqual(oldMask[i,j], mask[i,j])
+                    self.assertGreaterEqual(profile[i,j], 0.)
+
+            for i in range(spectrum.getLength()):
+                self.assertEqual(spectrum.getSpectrum()[i], spectrumFromProfile.getSpectrum()[i])
+
+        self.assertTrue(fiberTrace.createTrace(self.arc.getMaskedImage()))
+        self.assertEqual(oldHeight, fiberTrace.getHeight())
+        self.assertEqual(oldWidth, fiberTrace.getWidth())
+        self.assertEqual(oldProfile[5,5], fiberTrace.getProfile().getArray()[5,5])
+        self.assertNotAlmostEqual(oldTrace[5,5], fiberTrace.getTrace().getImage().getArray()[5,5])
+        spectrum = fiberTrace.extractFromProfile()
+        self.assertEqual(spectrum.getLength(), spectrumFromProfile.getLength())
+        self.assertNotAlmostEqual(spectrum.getSpectrum()[5], spectrumFromProfile.getSpectrum()[5])
 
     def testFiberTraceReconstruct(self):
-        if True:
-            fiberTraceSet = drpStella.findAndTraceAperturesF(self.flat.getMaskedImage(), self.ftffc)
-            self.ftffc.fiberTraceFunctionControl.xHigh = 6.0
-            self.assertFalse(
-                self.ftffc.fiberTraceFunctionControl is
-                fiberTraceSet.getFiberTrace(0).getFiberTraceFunction().fiberTraceFunctionControl
-            )
-            ftpfc = drpStella.FiberTraceProfileFittingControl()
-            ftpfc.profileInterpolation = "PISKUNOV"
-            for iTrace in range(0, fiberTraceSet.size()):
-                fiberTrace = fiberTraceSet.getFiberTrace(iTrace)
-                self.assertTrue(fiberTrace.setFiberTraceProfileFittingControl(ftpfc))
-                bool = fiberTrace.calcProfile()
-                self.assertTrue(bool)
+        fiberTraceSet = drpStella.findAndTraceAperturesF(self.flat.getMaskedImage(), self.ftffc)
+        self.ftffc.fiberTraceFunctionControl.xHigh = 6.0
+        self.assertFalse(
+            self.ftffc.fiberTraceFunctionControl is
+            fiberTraceSet.getFiberTrace(0).getFiberTraceFunction().fiberTraceFunctionControl
+        )
+        ftpfc = drpStella.FiberTraceProfileFittingControl()
+        ftpfc.profileInterpolation = "PISKUNOV"
+        for iTrace in range(0, fiberTraceSet.size()):
+            fiberTrace = fiberTraceSet.getFiberTrace(iTrace)
+            self.assertTrue(fiberTrace.setFiberTraceProfileFittingControl(ftpfc))
+            bool = fiberTrace.calcProfile()
+            self.assertTrue(bool)
 
-                spectrum = fiberTrace.extractFromProfile()
+            spectrum = fiberTrace.extractFromProfile()
 
-                recImage = fiberTrace.getReconstructed2DSpectrum(spectrum)
-                diff = fiberTrace.getTrace().getImage().getArray() - recImage.getArray()
-                if display:
-                    ds9.mtv(diff,title="reconstruction from MkSlitFunc",frame=fiberTraceSet.size()+iTrace)
```

Comment:
There should be *no* ds9 calls.  Please always use `afwDisplay`; PIPE2D-211

Out of scope for this ticket, but please note.